api/src/api/userRouter.ts

```diff
@@ -235,17 +245,35 @@ export const userRouterFactory = (
       return userData;
     }
     try {
-      const response = await timeStampBusinessSearch.search(userData.profileData.businessName);
+      const isForeign = userData.profileData.businessPersona === "FOREIGN";
+      const isDba = userData.profileData.needsNexusDbaName;
+      const nameToSearch = isDba ? userData.profileData.nexusDbaName : userData.profileData.businessName;
+      const response = await timeStampBusinessSearch.search(nameToSearch);
+
+      const nameAvailability: NameAvailability = {
+        status: response.status,
+        similarNames: response.similarNames ?? [],
+        lastUpdatedTimeStamp: response.lastUpdatedTimeStamp,
+        invalidWord: response.invalidWord,
+      };
+
+      const needsNexusDbaName =
+        isForeign && !isDba ? response.status !== "AVAILABLE" : userData.profileData.needsNexusDbaName;
```

Comment:
Again, if the variable assignment doesn't fit on a single line, it would be worth considering how we could break it down using temp variables maybe to make it more readable

Reply:
included